### PR TITLE
display pod lifecycle on intervals chart

### PR DESCRIPTION
**Pull request by deads2k**
This shows the basic pod and container lifecycle and the container readiness as reflected in the kube-api.

### CODE REVIEW

**Comment by deads2k** [comment]
/hold

holding until we see this work.

File: pkg/monitor/pod.go [hunk 1]
```diff
@@ -73,7 +73,39 @@ func startPodMonitoring(ctx context.Context, m Recorder, client kubernetes.Inter
 		return conditions
 	})
 
+	podScheduledFn := func(pod, oldPod *corev1.Pod) []monitorapi.Condition {
+		oldPodHasNode := oldPod != nil && len(oldPod.Spec.NodeName) > 0
+		newPodHasNode := pod != nil && len(pod.Spec.NodeName) > 0
+		if !oldPodHasNode && newPodHasNode {
+			return []monitorapi.Condition{
+				{
+					Level:   monitorapi.Warning,
```
Comment (dgoodwin) [inline pkg/monitor/pod.go:82]: Should being scheduled be a warning? Feels like info to me.

File: pkg/monitor/monitorapi/identification_pod.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+var (
+	// PodLifecycleTransitionReasons are the reasons associated with non-overlapping pod lifecycle states.
+	// A pod is logically identified by UID (I bet it's a name right now).
+	// Pods don't exist before create and don't exist after delete.
+	// Between those two states, each of these reasons can be ordered by time and used to create a contiguous view
+	// into the lifecycle of a pod.
+	PodLifecycleTransitionReasons = sets.NewString(
+		PodReasonCreated,
+		PodReasonScheduled,
```
Comment (dgoodwin) [inline pkg/monitor/monitorapi/identification_pod.go:111]: Is it feasible to cover Running as well? I found it strange looking at the html and seeing only Created/Scheduled, getting stuck there would indicate a problem.

**Review by dgoodwin — COMMENTED** [review]
I feel like this would be dramatically more useful if we could get the intervals for a specific pod onto the same line as we see with the alerts pending->firing. It's a bit difficult to see the lifecycle for a specific pod when you're searching for matching strings in that compressed mess on the right.

Other major question is around why you skipped Running.

Otherwise looks like this will be very useful.

File: pkg/monitor/monitorapi/identification_pod.go (hunk 2 above)
Comment (deads2k) [reply pkg/monitor/monitorapi/identification_pod.go:111]: > Is it feasible to cover Running as well? I found it strange looking at the html and seeing only Created/Scheduled, getting stuck there would indicate a problem.

Running is deceptive.  Pods have many containers and containers can be running, but the pod itself with a "running" phase doesn't tell you much useful.

Comment (dgoodwin) [reply pkg/monitor/monitorapi/identification_pod.go:111]: Does it tell us something with regards to not being stuck in Scheduled though? That the pod is doing "something", even if we don't know exactly what that is.

**Comment by deads2k** [comment]
1. broke node state change colors
2. container readiness isn't showing up
3. pods we missed create for, but which *are* scheduled are not showing as dark green (this will require memory in the monitor)
4. e2e pods are not excluded from kube-apiserver view
5. possibly worth creating new e2e interval that contains the pods mixed with tests
6. containerwaiting is entirely extraneous in the alerts view now
7. division by openshift namespace is possibly a better sort than time
8. pods that have runOnce containers where all the containers are finished are in a terminated state well before deletion.  The chart will be easier to read if terminated to deleted is in a light color

**Comment by deads2k** [comment]
> dramatically more useful if we could get the intervals for a specific pod onto the same line as we see with the alerts pending->firing.

I think we should hide the kubecontainerwaiting by default since we have a separate scheduling duration for pods.

**Comment by deads2k** [comment]
/hold

holding until I fix the punchlist above

**Pull request closed without merging**